solver/src/solver/solver_impl.rs

```diff
@@ -164,9 +164,59 @@ impl<Lbl: Label> Solver<Lbl> {
                 self.post_constraint(&equiv)
             }
             ReifExpr::Linear(lin) => {
-                assert!(self.model.entails(value), "Unsupported reified linear constraints.");
-                assert_eq!(self.model.presence_literal(value.variable()), Lit::TRUE);
-                self.reasoners.cp.add_linear_constraint(lin);
+                let lin = lin.cleaner();
+                let handled = match lin.sum.len() {
+                    0 => {
+                        // Check that the constant of the constraint is positive.
+                        self.post_constraint(&Constraint::Reified(
+                            ReifExpr::Lit(Lit::leq(VarRef::ZERO, lin.upper_bound)),
+                            value,
+                        ))?;
+                        true
+                    }
+                    1 => {
+                        let elem = lin.sum.first().unwrap();
+                        assert_ne!(elem.factor, 0);
+
+                        if lin.upper_bound % elem.factor != 0 {
+                            false
+                        } else {
+                            let v = lin.sum.first().unwrap().var;
+                            let lit = if elem.factor > 0 {
+                                SignedVar::plus(v)
+                            } else {
+                                SignedVar::minus(v)
+                            }
+                            .with_upper_bound(UpperBound::ub(lin.upper_bound / elem.factor));
+                            self.post_constraint(&Constraint::Reified(ReifExpr::Lit(lit), value))?;
+                            true
+                        }
+                    }
+                    2 => {
+                        // false
+                        let fst = lin.sum.get(0).unwrap();
+                        let snd = lin.sum.get(1).unwrap();
+                        assert_ne!(fst.factor, 0);
```

Comment:
These could be `debug_assert_ne!`: no matter what input is thrown at the function, this can never happen (unless you have a bug in `cleaner`).